server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -176,6 +180,38 @@ public RemoteSegmentMetadata initializeToSpecificCommit(long primaryTerm, long c
         return remoteSegmentMetadata;
     }
 
+    /**
+     * Initializes the remote segment metadata to a specific timestamp.
+     *
+     * @param timestamp The timestamp to initialize the remote segment metadata to.
+     * @return The RemoteSegmentMetadata object corresponding to the specified timestamp, or null if no metadata file is found for that timestamp.
+     * @throws IOException If an I/O error occurs while reading the metadata file.
+     */
+    public RemoteSegmentMetadata initializeToSpecificTimestamp(long timestamp) throws IOException {
+        List<String> metadataFiles = remoteMetadataDirectory.listFilesByPrefixInLexicographicOrder(
+            MetadataFilenameUtils.METADATA_PREFIX,
+            Integer.MAX_VALUE
+        );
+        Set<String> lockedMetadataFiles = RemoteStoreUtils.getPinnedTimestampLockedFiles(
```

Comment:
Inside `getPinnedTimestampLockedFiles` , we are checking for `RemoteStoreSettings.isPinnedTimestampsEnabled()` . However this can be called for pinned timestamps disabled clusters as well . 

There are more nuances to restoring across multiple clusters - the repo which stores pinned stamps should be known , pinned time stamp service should be initialized etc . We can discuss if we want to limit some knobs here to start simple - may be restore support on same domain with same repositories etc.

Reply:
> However this can be called for pinned timestamps disabled clusters as well .

Good point. As of now, the expectation is new cluster will have pinned timestamp enabled. Let me follow this up with a change where this restriction is not valid.

 > the repo which stores pinned stamps should be known

This may not be required as pinned timestamp is stored in the snapshot metadata.

Reply:
Should pinned timestamp be a per snapshot construct then?